File: leetcode/js/13.RomanToInteger.js
class Main{
    romanToInt(s){
      let helper = {
        'M':1000,
        'D':500,
        'C':100,
        'L':50,
        'X':10,
        'V':5,
        'I':1
      }
      var result =0
      for(let i=0;i<s.length;i++){
         if(i>0 && helper[s[i]] > helper[s[i-1]]){
            result -= 2 * helper[s[i-1]]
         }
         result += helper[s[i]]
      }
      return result
    }
  }
  
  var m = new Main()
  console.log(m.romanToInt("IV"))

File: leetcode/java/13.RomanToInteger.java
import java.util.*;

class Main{

public int romanToInt(String s){ 
  HashMap<Character,Integer> helper = new HashMap<Character,Integer>();
  initHelper(helper);
  int result = 0;
  for (int i = 0; i < s.length(); i ++){
    if ( i > 0 && helper.get(s.charAt(i)) > helper.get(s.charAt(i-1))) {
      result = result - 2 * helper.get(s.charAt(i-1));
    }
    result = result + helper.get(s.charAt(i));
  }
  return result;
}

public void initHelper(HashMap<Character,Integer> helper){
   helper.put('M',1000);
   helper.put('D',500);
   helper.put('C',100);
   helper.put('L',50);
   helper.put('X',10);
   helper.put('V',5);
   helper.put('I',1);
}
public static void main(String ...s){
    Main m = new Main();
    System.out.println(m.romanToInt("IV"));
} 
};
